pkg/kubectl/cmd/proxy.go

```diff
@@ -115,6 +115,7 @@ func RunProxy(f cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
 		AcceptPaths: kubectl.MakeRegexpArrayOrDie(cmdutil.GetFlagString(cmd, "accept-paths")),
 		RejectPaths: kubectl.MakeRegexpArrayOrDie(cmdutil.GetFlagString(cmd, "reject-paths")),
 		AcceptHosts: kubectl.MakeRegexpArrayOrDie(cmdutil.GetFlagString(cmd, "accept-hosts")),
+		RejectMethods:kubectl.MakeRegexpArrayOrDie(cmdutil.GetFlagString(cmd, "reject-methods")),
```

Comment:
Nice catch. Personally I think we should add this rejected methods as what we have already documented, this enables a more secure proxy server. It did break the currently things maybe we can add more clear error msg to tell users why you get rejected. wdyt?